src/librustc_mir/transform/simplify_cfg.rs

```diff
@@ -65,25 +44,70 @@ impl SimplifyCfg {
             Some(target)
         }
 
+        let mut predecessor_map = util::build_predecessor_map(mir);
+
         let mut changed = false;
-        for bb in mir.all_basic_blocks() {
+        let mut seen = vec![false; mir.basic_blocks.len()];
```

Comment:
It'd be interesting to instrument the compiler and find out what's the largest number of blocks we see in building `rustc` at least.